extensions/ql-vscode/src/variant-analysis/sarif-processing.ts

```diff
@@ -244,7 +254,7 @@ function getCodeFlows(
         for (const threadFlowLocation of threadFlow.locations) {
           const physicalLocation =
             threadFlowLocation!.location!.physicalLocation!;
```

Comment:
When we set `physicalLocation`, we're also making an assumption that it's provided and then calling `contextRegion` and `region` on it. Do we want to handle this better as well?

Reply:
I'm not really sure. There's a lot of this going on this file, where the SARIF spec says things can be undefined but then we ignore that and behave as if they're always defined. I don't want this PR to get too big, but I could open an issue to look into this more and see if we should make things stricter and safer.

Reply:
Makes sense. Up to you if you want to dig into it further. I just noticed it and wondered if it would also be a cause for crashes.